### ISSUE TITLE
Menu does not trap focus using NVDA with Chrome

### ISSUE THREAD

**Issue by jaredcrystal** [instructure/instructure-ui#1218]
### Background Information

**Package Version(s)**:
ui-menu v7.22.1

**Browser**:
Chrome

**OS:**
Windows 10

**Device:**
Windows PC

**Component**:
Menu

**Screen Reader:**
NVDA

### Describe the Issue
When rendered in an iframe, the Menu component fails to trap user focus when opened (using NVDA + Chrome).

### Steps To Reproduce
1. Using NVDA + Chrome
1. Go to https://instructure.design/v7/#Menu
1. Click the "Edit in Codepen" button under the example (This renders the example within the Codepen iframe)
1. Tab to the Menu trigger and use Space or Enter to expand the menu options
1. Use the Up or Down arrows until you begin interacting with elements outside the Menu (this should not be possible)

### Expected Behavior
The user's focus should be trapped within the menu options while open. Using the arrow keys should allow you to cycle through the options.

### Screenshots
https://drive.google.com/file/d/1ihbw_ull2Qei97XHgVzsyb9nAEypBcJH/view

### Additional Information

**Current Workaround(s):**
Using alt+up or alt+down while navigating the menu items traps focus correctly

**Products Affected:**
New Quizzes (filed as a critical bug https://instructure.atlassian.net/browse/QUIZ-11440)

**Requested Priority:**
Blocking (New Quizzes critical ticket blocked by this issue)

**Comment by matyasf** [comment]
hi!

I could reproduce the issue. Interestingly it does not show up in our doc site, but its there in Codepen. Is your code that exhibits this bug running in an iFrame?

**Comment by jaredcrystal** [comment]
Yes, this manifests in New Quizzes, which runs in an iframe inside of Canvas.

**Comment by matyasf** [comment]
We have determined that this is because reader mod in screenreders can escape iframes, this is working as intended
